src/_zkapauthorizer/_plugin.py

```diff
@@ -149,17 +153,28 @@ def _get_store(self, node_config):
             s = self._stores[key]
         except KeyError:
             s = open_store(datetime.now, _connect, node_config)
+            private_conn = _open_database(
+                partial(_connect, node_config.get_private_path(CONFIG_DB_NAME))
+            )
```

Comment:
I think the interfaces changes we made to `add_mutation_observer` and to call the resulting functions in `__exit__` outside of the transaction that methods starts means that we don't actually need this separate private connection.

So I think it would be good to drop this extra connection object.  A change that might go well with that is to make the underlying connection object available to this `_plugin.py` code though.  Very loosely:

```
unreplicated_conn = _open_database(...)
replicated_conn = with_replication(unreplicated_conn, is_replication_setup(node_config))
store = open_store(..., replicated_conn, ...)
self._add_replication_service(unreplicated_conn, replicated_conn, store, uploader)
```

This implies changes to `VoucherStore.from_connection` (basically moving the replication setup out of there) and I shuffled the argument order to `_add_replication_service` to make more sense to me).